scipy/stats/_multivariate.py

```diff
@@ -3589,18 +3589,11 @@ def rvs(self, dim, size=1, random_state=None):
 
         dim = self._process_parameters(dim)
 
-        H = np.eye(dim)
-        for n in range(dim):
-            x = random_state.normal(size=(dim-n,))
-            norm2 = np.dot(x, x)
-            x0 = x[0].item()
-            # random sign, 50/50, but chosen carefully to avoid roundoff error
-            D = np.sign(x[0]) if x[0] != 0 else 1
-            x[0] += D * np.sqrt(norm2)
-            x /= np.sqrt((norm2 - x0**2 + x[0]**2) / 2.)
-            # Householder transformation
-            H[:, n:] = -D * (H[:, n:] - np.outer(np.dot(H[:, n:], x), x))
-        return H
+        z = random_state.normal(size=(dim, dim))
+        q, r = scipy.linalg.qr(z)
+        d = r.diagonal()
+        q *= d/abs(d)
+        return q
```

Comment:
This is a copy-paste of the code from `unitary_group`, the only difference being the initialization of `z`. This is correct according to the discussion in gh-16560.